### PR TITLE
Implement feedbacks from Aestek:

### PR DESCRIPTION
**Pull request by tionebsalocin**
- Improve error management
- Fix argument management
- Display stdout and stderr of executed command

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- echo "Deploying $TRAVIS_TAG to GitHub releases"
+- tar czf consul-wrapper-linux.tar.gz consul-wrapper-linux
+- tar czf consul-wrapper-macos.tar.gz consul-wrapper-macos
+- zip -r consul-wrapper-windows.zip consul-wrapper-windows.exe
+deploy:
+  provider: releases
+  api_key:
+    secure: NsedjJtO6wzdnrWrX+c/8fxX8njUeNfSzWQqS/ggk9uKZD5dG2VcyRYzdVRbRqXEaiExuE/Id84xSzBZlKkdbBYJByvkZlyoPlr/w8OSkdJo++S9JhNkZWOSM7pjy9XTnGfk12K1TQAOf4JltHxwGRLsvazIOwpGrC9+suIXmxMHGpi73xGdwRRHRVvCrWwSIgFidqRgg4JTPRxhYNGEfRjiwTLnCf/SDK2gp7LlpeYsQqnYU094g00pZn2t/N5DrLtzwa+1jsJ7R8Tz1e6ubFwlI9jUYQ5r5jFrVAi0uTN6q3T+M6PJ6h/GfRp8XPbI28v+6gk0EIx/xXoOiuALXzaole8HETrHMHQwGxa8bvn1qRJT7NSN097oB2U3o9B61sYkyX7GFcmSerK2knKUv036eArirZEM4oSVMBB6GkZY6AuEq3fOyiTEcCPVt9ZGFiPVpfyCgvOHS0MKbaC8Zt9rlrrekeIqBOCepsKttms8y6hgn8cjMkugN3sRLrZWeeX3GL8QutZWECnKlXwA660oE6IR4V7Uf27jWBwZ8ULcU1vEu0OmyLEnn6nNXnX3SsRmqsxoqZNhnUKXWTz51JJog1cyyelevY2tNS8IjWy953Jzh1Ps46NUNDDSFOMCGHN7E80B5CVnNj/Iac1fzXN/4zmL5jtqV+69UlRMSzk=
+  file_glob: true
+  file: consul-wrapper-*.tar.gz
```
Comment (ShimmerGlass) [inline .travis.yml]: won't release the windows zip file

File: main.go [hunk 2]
```diff
@@ -37,50 +36,71 @@ func getConsulClient(httpAddr string, token string) *api.Client {
...
+	log.Printf("[ConsulWrapper] Deregistering '%s' from consul\n", name)
+	err := client.Agent().ServiceDeregister(name)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "Usage of %s:\n\n", os.Args[0])
+	fmt.Fprintf(os.Stderr, "\t%s [options] -service <ServiceName> -command <Command> [Args]\n\n", os.Args[0])
```
Comment (ShimmerGlass) [inline main.go]: You can use `flag.PrintDefaults()`

File: main.go [hunk 3]
```diff
@@ -37,50 +36,71 @@ func getConsulClient(httpAddr string, token string) *api.Client {
...
+	flag.PrintDefaults()
 }
 
 func main() {
 	tokenPrt := flag.String("token", "", "Consul token used for registration")
 	servicePtr := flag.String("service", "", "Consul Service Name")
-	checkFrequencyPtr := flag.Int("frequency", 30, "Health Check Frequency (in seconds)")
+	checkFrequencyPtr := flag.Duration("frequency", time.Duration(30), "Health Check Frequency (in seconds)")
 	commandPtr := flag.String("command", "", "Command to run")
```
Comment (ShimmerGlass) [inline main.go]: You can remove this and just use `flag.Args()`

File: main.go [hunk 4]
```diff
@@ -37,50 +36,71 @@ func getConsulClient(httpAddr string, token string) *api.Client {
...
+	log.Printf("[ConsulWrapper] Registering '%s' in consul\n", name)
+	err := client.Agent().ServiceRegister(&api.AgentServiceRegistration{
 		ID:   name,
 		Name: name,
 		Check: &api.AgentServiceCheck{
 			CheckID:                        name,
 			Name:                           name,
-			DeregisterCriticalServiceAfter: strconv.Itoa(2*frequency) + "s",
-			TTL:                            strconv.Itoa(3*frequency) + "s",
+			DeregisterCriticalServiceAfter: fmt.Sprintf("%d", int64(2.1*frequency.Seconds())) + "s",
```
Comment (ShimmerGlass) [inline main.go:53]: You can use 
```
(2.1*frequency).String()
```

File: main.go [hunk 5]
```diff
@@ -37,50 +36,71 @@ func getConsulClient(httpAddr string, token string) *api.Client {
...
+			TTL:                            fmt.Sprintf("%d", int64(3.1*frequency.Seconds())) + "s",
```
Comment (ShimmerGlass) [inline main.go:54]: Same

File: main.go [hunk 6]
```diff
@@ -6,22 +6,21 @@ import (
...
 		_, err := os.FindProcess(pid)
 		if err != nil {
-			fmt.Println("Process is not running")
+			log.Println("[ConsulWrapper] Process is not running")
 			client.Agent().FailTTL(checkName, "Process is not running")
 			break
 		}
-		fmt.Println("Process is running")
+		log.Println("[ConsulWrapper] Process is running")
 		client.Agent().PassTTL(checkName, "Process is running")
```
Comment (ShimmerGlass) [inline main.go]: err check

File: main.go [hunk 7]
```diff
@@ -6,22 +6,21 @@ import (
...
-func processWatcher(pid int, checkName string, frequency int, client *api.Client) {
-	channel := time.Tick(time.Duration(frequency) * time.Second)
+func processWatcher(pid int, checkName string, frequency time.Duration, client *api.Client) {
+	channel := time.Tick(frequency)
 	for range channel {
 		_, err := os.FindProcess(pid)
 		if err != nil {
-			fmt.Println("Process is not running")
+			log.Println("[ConsulWrapper] Process is not running")
 			client.Agent().FailTTL(checkName, "Process is not running")
```
Comment (ShimmerGlass) [inline main.go]: err check

**Review by ShimmerGlass — CHANGES_REQUESTED** [review]

File: main.go (hunk 3 above)
Comment (tionebsalocin) [reply main.go]: Done

File: main.go (hunk 2 above)
Comment (tionebsalocin) [reply main.go]: The argument part would not be displayed...

File: main.go (hunk 4 above)
Comment (tionebsalocin) [reply main.go:53]: it won't fit in this case due to .1

File: main.go (hunk 5 above)
Comment (tionebsalocin) [reply main.go:54]: same

File: main.go (hunk 6 above)
Comment (tionebsalocin) [reply main.go]: Fixed

File: main.go (hunk 7 above)
Comment (tionebsalocin) [reply main.go]: fixed

File: .travis.yml (hunk 1 above)
Comment (tionebsalocin) [reply .travis.yml]: Fixed

**Comment by ShimmerGlass** [comment]
I'm not sure we should crash if the `client.Agent().{Pass,Fail}TTL()` call fail. If we loose leadership all apps would stop.

**Pull request merged**